File: pkg/runtime/runtimes/python2/pulserpc/rpc.py
"""RPC error handling for JSON-RPC 2.0"""


class RPCError(Exception):
    """Exception class for JSON-RPC errors"""
    
    def __init__(self, code, message, data=None):
        self.code = code
        self.message = message
        self.data = data
        super(RPCError, self).__init__("RPCError %d: %s" % (code, message))

File: pkg/runtime/runtimes/ts-cjs/pulserpc/rpc.ts
/**
 * RPC error handling for JSON-RPC 2.0
 */

export class RPCError extends Error {
  code: number;
  data?: any;

  constructor(code: number, message: string, data?: any) {
    super(message);
    this.name = `RPCError ${code}`;
    this.code = code;
    this.data = data;
    if (typeof (Error as any).captureStackTrace === "function") {
      (Error as any).captureStackTrace(this, RPCError);
    }
  }
}
